File: LeetCode/1877数组中最大数对和的最小值.cpp
int cmp(const void *a, const void *b) {
    return *(int *)a - *(int *)b;
}

int minPairSum(int* nums, int numsSize){
    qsort(nums, numsSize, sizeof(int), cmp);
    int max = nums[0] + nums[numsSize - 1];
    for (int i = 1; i < numsSize / 2; i++) {
        max = fmax(max, nums[i] + nums[numsSize - i - 1]);
    }
    return max;
}

File: LeetCode/1877数组中最大数对和的最小值.go
const INT_MAX = int(^uint32((0)) >> 1)
func minPairSum(nums []int) int {
    sort.Sort(sort.IntSlice(nums))
    ans := 0
    for i := 0; i < len(nums) / 2; i++ {
        ans = int(math.Max(float64(nums[i] + nums[len(nums) - 1 - i]), float64(ans)))
    }
    return ans
}